PreviewViewControllerExample/SceneDelegate.swift

```diff
@@ -0,0 +1,164 @@
+    func setupGestureRecognizers() {
+        let longPressGestureRecognizer = UILongPressGestureRecognizer(target: self, action: #selector(handleLongPress(_:)))
+        previewViewController.navigationMapView.addGestureRecognizer(longPressGestureRecognizer)
+    }
+    
+    @objc func handleLongPress(_ gesture: UILongPressGestureRecognizer) {
+        guard gesture.state == .ended,
+              let passiveLocationProvider = previewViewController.navigationMapView.mapView.location.locationProvider as? PassiveLocationProvider,
+              let originCoordinate = passiveLocationProvider.locationManager.location?.coordinate else { return }
+        
+        let destinationCoordinate = previewViewController.navigationMapView.mapView.mapboxMap.coordinate(for: gesture.location(in: previewViewController.navigationView.navigationMapView.mapView))
+        let coordinates = [
+            originCoordinate,
+            destinationCoordinate,
+        ]
+        
+        let topmostBottomBanner = previewViewController.topBanner(at: .bottomLeading)
+        
+        // In case if `RoutesPreviewViewController` is shown - don't do anything.
+        if topmostBottomBanner is RoutesPreviewViewController {
+            return
+        }
+        
+        // In case if `DestinationPreviewViewController` is shown - dismiss it and after that show new one.
+        if topmostBottomBanner is DestinationPreviewViewController {
+            previewViewController.dismissBanner(at: .bottomLeading,
+                                                animated: false)
+            preview(coordinates,
+                    animated: false)
+        } else {
+            if shouldAnimate {
+                previewViewController.navigationView.topBannerContainerView.alpha = 0.0
+                previewViewController.navigationView.bottomBannerContainerView.alpha = 0.0
+            }
+            
+            preview(coordinates,
+                    animated: shouldAnimate,
+                    duration: animationDuration,
+                    animations: { [self] in
+                self.previewViewController.navigationView.topBannerContainerView.alpha = 1.0
+                self.previewViewController.navigationView.bottomBannerContainerView.alpha = 1.0
+            })
+        }
+    }
+    
+    func presentBannerDismissalViewControllerIfNeeded(_ animated: Bool,
+                                                      duration: TimeInterval) {
+        if previewViewController.topBanner(at: .topLeading) is BannerDismissalViewController {
+            return
+        }
+        
+        let bannerDismissalViewController = BannerDismissalViewController()
+        bannerDismissalViewController.delegate = self
+        previewViewController.present(bannerDismissalViewController,
+                                      animated: animated,
+                                      duration: duration)
+    }
+    
+    func preview(_ coordinates: [CLLocationCoordinate2D],
```

Comment:
I try to make `PreviewViewController` as light as possible in first iteration. Methods that allow to preview destination and routes can potentially move to it at some point.